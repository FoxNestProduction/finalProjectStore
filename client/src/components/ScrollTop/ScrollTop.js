import React from 'react';
import { useScrollTrigger } from '@mui/material';
import Fade from '@mui/material/Fade';
import Box from '@mui/material/Box';
import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp';
import Fab from '@mui/material/Fab';

const ScrollTop = () => {
  const trigger = useScrollTrigger({
    disableHysteresis: true,
    threshold: 300,
  });

  const handleClick = () => {
    window.scrollTo(0, 0);
  };

  return (
    <Fade in={trigger}>
      <Box
        onClick={handleClick}
        sx={{ position: 'fixed', bottom: 18, right: 18 }}
      >
        <Fab size="small" aria-label="scroll back to top">
          <KeyboardArrowUpIcon />
        </Fab>
      </Box>
    </Fade>
  );
};

export default ScrollTop;
