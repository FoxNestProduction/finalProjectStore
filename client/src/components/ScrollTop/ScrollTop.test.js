import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import ScrollTop from './ScrollTop';

describe('ScrollTop component', () => {
  beforeEach(() => {
    window.scrollTo = jest.fn();
  });

  afterEach(() => {
    window.scrollTo.mockRestore();
  });

  test('should render ScrollTop component', () => {
    const { asFragment } = render(<ScrollTop />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('ScrollTop component should scroll back to top when clicked', () => {
    render(<ScrollTop />);

    const scrollButton = screen.getByLabelText('scroll back to top');

    fireEvent.click(scrollButton);

    expect(window.scrollTo).toHaveBeenCalledWith(0, 0);
  });
});
