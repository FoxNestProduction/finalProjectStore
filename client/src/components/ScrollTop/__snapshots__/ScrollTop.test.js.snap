// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollTop component should render ScrollTop component 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-zxtiae"
    style="opacity: 0; visibility: hidden;"
  >
    <button
      aria-label="scroll back to top"
      class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default css-q288u8-MuiButtonBase-root-MuiFab-root"
      tabindex="0"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="KeyboardArrowUpIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</DocumentFragment>
`;
